js/script.js

```diff
@@ -0,0 +1,90 @@
+//Event listeners
+button.addEventListener("click", inquiryRequest);
+document.addEventListener("keypress", enableEnterKey);
+userInquiry.addEventListener("click", function () {
+  resetInput();
+});
+
+//Functions
+window.onload = createMarquee();
+
+function inquiryRequest() {
+  if (userInquiry.value.length > 0) {
+    listDisplay.innerHTML = "";
+    toggleList();
+    toggleLoader();
+    return getNasdaqStats(userInquiry.value);
+  }
+  if (userInquiry.value.length > 0 && list === true) {
+    toggleLoader();
+    return getNasdaqStats(userInquiry.value);
+  }
+  toggleList();
+  listDisplay.innerHTML = "";
+}
+
+const getNasdaqStats = async (userQuery) => {
+  const url = `${Url.searchRequest}/search?query=${userQuery}&limit=10&exchange=NASDAQ`;
+  listDisplay.classList.remove("error");
+  listDisplay.classList.add("listItem");
+  try {
+    let response = await fetch(url);
+    response = await response.json();
+    if (response.length === 0) {
+      throw "No matches found";
+    }
+    createList(response);
+    resetInput();
+  } catch (error) {
+    listDisplay.classList.add("error");
+    listDisplay.innerText = error;
+    toggleLoader();
+  }
+};
+
+const createList = async (data) => {
+  try {
+    let listContainer = document.createElement("div");
+    listContainer.classList.add("listContainer");
+    listContainer.style.display = "none";
+    listDisplay.appendChild(listContainer);
+    for (element in data) {
+      let itemContainer = document.createElement("span");
+      let queryResult = data[element];
+      let companyInfo = await recieveFurtherData(queryResult.symbol);
```

Comment:
this line delays the operation in a lot of important seconds. Promise.all() will solve it.